api/v1beta1/types.go

```diff
@@ -107,6 +107,97 @@ type Network struct {
 	APIInternalForwardingRule *string `json:"apiInternalForwardingRule,omitempty"`
 }
 
+// FirewallDescriptor describes a GCP firewall rule.
+type FirewallDescriptor struct {
+	// IPProtocol is the IP protocol to which this rule applies. The protocol type is
+	// required when creating a firewall rule. This value can either be one of the
+	// following well known protocol strings (tcp, udp, icmp, esp, ah, ipip, sctp)
+	// or the IP protocol number.
+	IPProtocol string `json:"IPProtocol,omitempty"`
+	// Ports is an optional list of ports to which this rule applies. This field is
+	// only applicable for the UDP or TCP protocol. Each entry must be either an
+	// integer or a range. If not specified, this rule applies to connections
+	// through any port. Example inputs include: ["22"], ["80","443"], and
+	// ["12345-12349"].
+	Ports []string `json:"ports,omitempty"`
+}
+
+// FirewallRule describes a GCP firewall rule.
+type FirewallRule struct {
+	// Allowed is the list of ALLOW rules specified by this firewall. Each rule
+	// specifies a protocol and port-range tuple that describes a permitted
+	// connection.
+	Allowed []*FirewallDescriptor `json:"allowed,omitempty"`
+	// Denied is the list of DENY rules specified by this firewall. Each rule
+	// specifies a protocol and port-range tuple that describes a denied
+	// connection.
+	Denied []*FirewallDescriptor `json:"denied,omitempty"`
+	// Description is an optional description of this resource. Provide this field
+	// when you create the resource.
+	Description *string `json:"description,omitempty"`
+	// DestinationRanges is the list of IP addresses where traffic applies. An empty
+	// list means all destinations. These ranges must be expressed in CIDR format.
+	// Both IPv4 and IPv6 are supported.
+	DestinationRanges []string `json:"destinationRanges,omitempty"`
```

Comment:
We should make sure we validate that these are real IP addresses? What is the minimum version of Kube that we support here? Perhaps we can now use CEL to validate that these are valid IP addresses